File: android/lunarcore/src/test/kotlin/com/lunarcore/SolarTermTests.kt
package com.lunarcore

import kotlin.test.Test
import kotlin.test.assertFalse
import kotlin.test.assertTrue

class SolarTermTests {

    @Test
    fun tietKhiNonEmpty() {
        val engine = LunarEngine()
        val solar = SolarDate(2024, 3, 20)
        val term = engine.tietKhi(solar)
        assertFalse(term.isEmpty())
    }

    @Test
    fun aroundLapXuan() {
        val engine = LunarEngine()
        val solar = SolarDate(2024, 2, 4)
        val term = engine.tietKhi(solar)
        assertTrue(
            term == "Lập Xuân" || term == "Vũ Thủy" || term == "Đại Hàn"
        )
    }
}

File: ios/LunarCore/Tests/LunarCoreTests/SolarTermTests.swift
//
//  SolarTermTests.swift
//  LunarCoreTests
//
//  Solar term (Tiết Khí) correctness.
//

import Testing
import LunarCore

@Suite("Solar terms")
struct SolarTermTests {

    @Test("Tiết Khí returns non-empty name")
    func tietKhiNonEmpty() {
        let engine = LunarEngine()
        let solar = SolarDate(year: 2024, month: 3, day: 20)
        let term = engine.tietKhi(date: solar)
        #expect(!term.isEmpty)
    }

    @Test("Around Xuân Phân (March ~20)")
    func xuanPhan() {
        let engine = LunarEngine()
        let solar = SolarDate(year: 2024, month: 3, day: 20)
        let term = engine.tietKhi(date: solar)
        #expect(term == "Xuân Phân" || term == "Thanh Minh" || term == "Cốc Vũ")
    }

    @Test("Around Lập Xuân (Feb ~4)")
    func lapXuan() {
        let engine = LunarEngine()
        let solar = SolarDate(year: 2024, month: 2, day: 4)
        let term = engine.tietKhi(date: solar)
        #expect(term == "Lập Xuân" || term == "Vũ Thủy" || term == "Đại Hàn")
    }
}
